### PR TITLE
Clarify API GetDatabases vs GetDatabaseServers

### PR DESCRIPTION
**Pull request by greedy52**
Was asked why `GetDatabases` returns 0 databases and doesn't match `tsh db ls` during support shift.

### CODE REVIEW

File: api/client/client.go [hunk 1]
```diff
@@ -2630,6 +2634,14 @@ func (c *Client) UpdateApp(ctx context.Context, app types.Application) error {
...
 
 // GetApp returns the specified application resource.
+//
+// Note that application resources here refers to "dynamically-added"
+// applications such as applications created by `tctl create`, or the CreateApp
+// API. Applications defined in the `app_service.apps` section of the service
+// yaml configuration are not collected in this API.
+//
+// For a full list of registered applications that are served by an application
+// service service, use GetApplicationServers instead.
```
Comment (zmb3) [inline api/client/client.go:2644]:
```suggestion
// service, use GetApplicationServers instead.
```

similarly

File: api/client/client.go [hunk 2]
```diff
@@ -2630,6 +2634,14 @@ func (c *Client) UpdateApp(ctx context.Context, app types.Application) error {
 }
 
 // GetApp returns the specified application resource.
+//
+// Note that application resources here refers to "dynamically-added"
+// applications such as applications created by `tctl create`, or the CreateApp
+// API. Applications defined in the `app_service.apps` section of the service
+// yaml configuration are not collected in this API.
```
Comment (zmb3) [inline api/client/client.go:2641]:
```suggestion
// YAML configuration are not collected in this API.
```

(here and below)

**Review by zmb3 — APPROVED** [review]

**Review by jakule — APPROVED** [review]
LGTM, but should we also explain the differences in our official docs? I don't think that too many people read GoDocs when they are looking for answers.

**Review by smallinsky — APPROVED** [review]

**Comment by greedy52** [comment]
> LGTM, but should we also explain the differences in our official docs? I don't think that too many people read GoDocs when they are looking for answers.

The API doc https://goteleport.com/docs/api/getting-started/ sends users to https://pkg.go.dev/github.com/gravitational/teleport/api/client

Any suggestions other doc references we want to update?

**Pull request merged**